src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -555,13 +555,16 @@ internal bool IsInitialized(Guid guid)
         /// <returns>The custom node info object - null if we failed</returns>
         internal bool TryGetInfoFromPath(string path, bool isTestMode, out CustomNodeInfo info)
         {
+            WorkspaceInfo header = null;
             try
             {
-                var xmlDoc = new XmlDocument();
-                xmlDoc.Load(path);
+                XmlDocument xmlDoc;
+                string jsonDoc;
+                DynamoUtilities.PathHelper.isValidXML(path, out xmlDoc);
+                DynamoUtilities.PathHelper.isValidJson(path, out jsonDoc);
 
-                WorkspaceInfo header;
-                if (!WorkspaceInfo.FromXmlDocument(xmlDoc, path, isTestMode, false, AsLogger(), out header))
+                if (!WorkspaceInfo.FromXmlDocument(xmlDoc, path, isTestMode, false, AsLogger(), out header)
+                && !WorkspaceInfo.FromJsonDocument(jsonDoc, path, isTestMode, false, AsLogger(), out header))
```

Comment:
isValidXML and isValidJson  should be used here before using FromJsonDocument or FromXmlDocument.

Reply:
@ramramps how can I fix it?
`newWorkspace = (CustomNodeWorkspaceModel)WorkspaceModel.FromJson(jsonDoc, null, null, null, nodeFactory, false, true, this);`
I have received an exception because LibraryServices is null

Reply:
@phliberato Would you give me some time on this? I looked at the Json deserialization and found bigger holes in it. I feel like we should not need to pass in so many params and could use only nodeFactory.

Reply:
@QilongTang thanks for your support. I can wait for it. If you need somebody to tests this PR you may contact me.